yes.rs

```diff
@@ -1,19 +1,11 @@
 use std::env;
 
-#[allow(while_true)]
 fn main() {
     let args: Vec<String> = env::args().collect();
+    let args = &args[1..];
 
-    match args.len() {
-        1 => {
-            while true {
-                println!("y");
-            }
-        },
-        _ => {
-            while true {
-                println!("{}", args[1]);
-            }
-        },
+    let output = if args.is_empty() { "y".to_string() } else { args.join(" ") };
```

Comment:
@RobertZenz yes. Basically you need expressions of the same type in both branches. String literals have type `&str`, and `args.join` returns String. I actually fought with borrow checker here a bit, I failed to find a way to get `&str` from `args.join` and do it the other way.